subsys/net/lib/websocket/websocket.c

```diff
@@ -419,6 +423,52 @@ static int websocket_interal_disconnect(struct websocket_context *ctx)
 
 	NET_DBG("[%p] Disconnecting", ctx);
 
+	ret = websocket_send_msg(ctx->sock, NULL, 0, WEBSOCKET_OPCODE_CLOSE,
+				 true, true, 0);
+	if (ret < 0) {
+		NET_ERR("[%p] Failed to send close message (err %d).", ctx, ret);
+	}
+
+	fds.fd = ctx->real_sock;
+	fds.events = ZSOCK_POLLIN;
+
+	/* Arbitrary timeout value for waiting for close reply. */
+	start = k_uptime_get();
+	timeout = 2000;
+	while (timeout > 0) {
```

Comment:
BTW, if the peer does not send a reply to the close, then we have a timeout. Just wondering if this is desired or not, should we just ignore the reply in order not to get into a timeout case, wdyt?

Reply:
I'm not sure I follow, do you suggest to skip the `zsock_poll()`/`websocket_recv_msg()` and then close the socket immediately after sending close message? I'm not sure if we'll be RFC compliant then, specifically:
> the client SHOULD wait for the server to
   close the connection but MAY close the connection at any time after
   sending **and receiving a Close message**

So in ideal conditions, the client should wait for the reply. Note, that if the server decides to close the underlying connection, the `zsock_poll()` will exit immediately.

Reply:
I was just reading your comment in #53942 where you mentioned that not many servers support the close message. So I am just pondering would that cause some issues with extra waits if in our side we expect a reply.

Reply:
Well, if the server is completely out-of-spec, we'd just waste 2 seconds. But those servers that I mentioned, although did not send WebSocket close response, closed the underlying TCP connection immediately, so we're not waiting on our side.